src/multi.c

```diff
@@ -289,6 +289,8 @@ void watchForKey(client *c, robj *key) {
     wk = zmalloc(sizeof(*wk));
     wk->key = key;
     wk->db = c->db;
+    /* Use the lru field to flag that we're watching an already expired key. */
+    key->lru = keyIsExpired(c->db, key);
```

Comment:
not a fan of this hack. let's consider a bit in `watchedKey` struct.
maybe change `db` pointer to `db` index, so that we have spare bits.

Reply:
> change `db` pointer to `db` index, so that we have spare bits

That's a great idea!

Reply:
> maybe change `db` pointer to `db` index, so that we have spare bits.

`touchAllWatchedKeysInDb` is called from `swapMainDbWithTempDb` with a tempDb that is not in the `server.db` array, so using index instead of pointer doesn't work in this case. I'm reverting to a db pointer in watchedKey.